features/upgrade/sdn/policy-upgrade.feature

```diff
@@ -233,3 +233,121 @@ Feature: SDN compoment upgrade testing
+    When I run the :create admin command with:
+      | f | allow-from-router-ingress.yaml     |
+    Then the step should succeed
+
+    When I use the "policy-upgrade4" project
+    Given I obtain test data file "networking/list_for_pods.json"
+    When I run oc create over "list_for_pods.json" replacing paths:
+      | ["items"][0]["spec"]["replicas"]  | 1    |
+    Then the step should succeed
+    Given a pod becomes ready with labels:
+      | name=test-pods |
+    When I expose the "test-service" service
+    And I wait up to 60 seconds for a web server to become available via the "test-service" route
+    Then the step should succeed
+    Given the DefaultDeny policy is applied to the "policy-upgrade4" namespace
+    Then the step should succeed
+    Given I obtain test data file "networking/networkpolicy/allow-from-router.yaml"
+    When I run the :create admin command with:
+      | f | allow-from-router.yaml     |
+    Then the step should succeed
+
+    When I use the "policy-upgrade5" project
+    Given I obtain test data file "networking/list_for_pods.json"
+    When I run oc create over "list_for_pods.json" replacing paths:
+      | ["items"][0]["spec"]["replicas"]  | 1    |
+    Then the step should succeed
+    Given a pod becomes ready with labels:
+      | name=test-pods |
+    Then the step should succeed
+    And evaluation of `pod(2).ip_url` is stored in the :p5pod1ip clipboard
+    Given I save multus pod on master node to the :multuspod clipboard
+    Given the DefaultDeny policy is applied to the "policy-upgrade5" namespace
+    Then the step should succeed
+    Given I obtain test data file "networking/networkpolicy/allow-from-hostnetwork.yaml"
+    When I run the :create admin command with:
+      | f | allow-from-hostnetwork.yaml     |
+    Then the step should succeed
+
+    #Validation
+    And I wait up to 30 seconds for the steps to pass:
+    """
+    When I use the "policy-upgrade3" project
+    When I open web server via the "test-service" route
+    Then the step should succeed
+
+    When I use the "policy-upgrade4" project
+    When I open web server via the "test-service" route
+    Then the step should succeed
+
+    Given I switch to cluster admin pseudo user
+    Given I use the "openshift-multus" project
+    When I execute on the "<%= cb.multuspod %>" pod:
+      | curl | -I | <%= cb.p5pod1ip %>:8080 |
+    Then the step should succeed
+    And the output should contain "200 OK"
+    """
+
+  # @author asood@redhat.com
+  # @case_id OCP-40620
+  @admin
+  @upgrade-check
+  Scenario: Check allow from router and allow from hostnetwork policy are functional post upgrade 
+    Given I switch to cluster admin pseudo user
+    When I use the "policy-upgrade3" project
+    Given a pod becomes ready with labels:
+      | name=test-pods |
+    And I wait up to 60 seconds for a web server to become available via the "test-service" route
+    Then the step should succeed
```

Comment:
@asood-rh if the previous `And I wait up to 60...` step checks for success for failure, then you don't need to post check `Then the step should succeed`

Reply:
@asood-rh so for that [step I mentioned ](https://github.com/openshift/verification-tests/blob/master/features/step_definitions/net.rb#L37) above it indeed checks for success/failure, so I think we can remove line 333.  The same argument applies to all other steps that exhibit that kind of behavior.  Even though in cucumber the key words `GIven/And/Then` are treated equally, semantic-wise, steps using `Given` implies the status check is embedded.

Reply:
@pruan-rht Thanks for review. 
Removed the suggested post checks. Reviewed other post checks as well, they are needed.
https://mastern-jenkins-csb-openshift-qe.apps.ocp4.prod.psi.redhat.com/job/Runner-v3-smoke/3270/console
https://mastern-jenkins-csb-openshift-qe.apps.ocp4.prod.psi.redhat.com/job/Runner-v3-smoke/3271/console